File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TJSONNull.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TJSONNull.h"


@implementation TJSONNull
-(NSString *) asJSONString{
	return @"null";
}
-(id) getInternalObject {
	return [NSNull null];
}
-(NSString *) toString{
	return [self asJSONString];
}
-(JSONValueType) getJSONValueType{
	return JSONNull;
}


@end

File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TJSONNull.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

import org.json.me.bc.JSONObject;

/**
 * Contains and returns the 'null' value like a JSON .
 */

public class TJSONNull extends TJSONValue {

	public static Object Null = JSONObject.NULL;
	
	public int getJsonValueType() {
		return JSONValueType.JSONNull;
	}

	/**
	 * Returns the String "null"
	 * 
	 * @return
	 */
	public String asJSONString() {
		return "null";
	}

	public Object getInternalObject() {
		return JSONObject.NULL;
	}

	public String toString() {
		return asJSONString();
	}
}